src/main/java/com/uid2/shared/store/parser/ServiceParser.java

```diff
@@ -0,0 +1,37 @@
+package com.uid2.shared.store.parser;
+
+import com.uid2.shared.Utils;
+import com.uid2.shared.auth.Role;
+import com.uid2.shared.model.Service;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+
+public class ServiceParser implements Parser<Map<Integer, Service>> {
+
+    @Override
+    public ParsingResult<Map<Integer, Service>> deserialize(InputStream inputStream) throws IOException {
+        JsonArray spec = Utils.toJsonArray(inputStream);
+        final HashMap<Integer, Service> serviceMap = new HashMap<>();
+        for (int i = 0; i < spec.size(); i++) {
+            JsonObject serviceSpec = spec.getJsonObject(i);
+            int serviceId = serviceSpec.getInteger("service_id");
+            int siteId = serviceSpec.getInteger("site_id");
+            String name = serviceSpec.getString("name");
+            JsonArray rolesSpec = serviceSpec.getJsonArray("roles");
```

Comment:
What happens if rolesSpec is null? Can it be?

Reply:
I have added some logic into Service.java to enforce non-null. The jsonWriter used in uid2-admin will write empty set as [ ]

Reply:
If roles is not specified in the json, the call roleSpec.size below will throw an exception

Reply:
Roles should be required to be specified in the json like the other fields, but we can allow it to be empty json array.